lib/sparse/src/index/posting_list.rs

```diff
@@ -86,13 +105,7 @@ impl PostingList {
         for element in self.elements[..up_to_index].iter_mut().rev() {
             // update max_next_weight for element
             element.max_next_weight = max_next_weight;
-            if element.weight >= max_next_weight {
```

Comment:
why did you change the logic here?

Reply:
because not we need to not only propagate weights when they are inserted (increase only) but also when they are removed (decrease). But this logic is up to removal anyway as per discussion here https://github.com/qdrant/qdrant/pull/4375#discussion_r1623000524